internal/db/notification.go

```diff
@@ -9,30 +9,39 @@ import (
 	"github.com/slovak-egov/einvoice/pkg/context"
 )
 
-func (c *Connector) GetNotNotifiedInvoices(ctx goContext.Context, limit int) ([]entity.Invoice, error) {
+func (c *Connector) GetAndUpdateNotNotifiedInvoices(ctx goContext.Context, limit int) ([]entity.Invoice, error) {
 	invoices := []entity.Invoice{}
-	query := c.GetDb(ctx).Model(&invoices).
-		Column("id", "customer_ico", "supplier_ico").
-		Where("notifications_sent = FALSE").
+	notUpdatedInvoices := c.GetDb(ctx).
+		Model(&entity.Invoice{}).
+		Column("id").
+		Where("notifications_status = 'not_sent'").
 		Order("id ASC").
-		Limit(limit)
+		Limit(limit).
+		For("UPDATE SKIP LOCKED")
+
+	query := c.GetDb(ctx).
+		Model(&invoices).
+		Set("notifications_status = 'sending'").
+		With("ids", notUpdatedInvoices).
+		Where("id IN (SELECT * FROM ids)").
+		Returning("*")
```

Comment:
we probably dont need to return everything, `id` and parties `ico` should be sufficient